### PR TITLE
ATO-627: Update AIS tests to cover case that fields are missing in AIS response

### PR DESCRIPTION
**Pull request by billrdunn**
## What

AIS may not send us all the fields under "intervention" in the response. This PR modifies the contract test to let AIS know we do not need these fields, and adds unit and integration tests to cover this situation.

## Checklist

- [x] Impact on orch and auth mutual dependencies has been checked.

### CODE REVIEW

File: oidc-api/src/test/java/uk/gov/di/authentication/oidc/contract/AccountInterventionServiceTest.java [hunk 1]
```diff
@@ -358,12 +358,7 @@ private static DslPart createAisResponseBody(
                             body.object(
                                     "intervention",
                                     (obj) -> {
-                                        obj.numberType("updatedAt", TIME_NOW);
                                         obj.numberType("appliedAt", TIME_NOW);
-                                        obj.numberType("sentAt", TIME_NOW);
-                                        obj.stringType("description", "TEST_DESCRIPTION");
-                                        obj.numberType("reprovedIdentityAt", TIME_NOW);
-                                        obj.numberType("resetPasswordAt", TIME_NOW);
                                     });
```
Comment (isaac-GDS) [inline oidc-api/src/test/java/uk/gov/di/authentication/oidc/contract/AccountInterventionServiceTest.java:362]: Is the default to return only "appliedAt". Do we care about the other fields at any point in the `intervention` object ?

Comment (billrdunn) [reply oidc-api/src/test/java/uk/gov/di/authentication/oidc/contract/AccountInterventionServiceTest.java:362]: No the default is to return either all the fields or all the fields excluding `reprovedIdentityAt` and `resetPasswordAt`. The only field we use is `appliedAt` and we don't care about any other fields

Comment (isaac-GDS) [reply oidc-api/src/test/java/uk/gov/di/authentication/oidc/contract/AccountInterventionServiceTest.java:362]: Cool - this is fine

File: shared-test/src/main/java/uk/gov/di/authentication/sharedtest/extensions/AccountInterventionsStubExtension.java [hunk 2]
```diff
@@ -91,6 +91,39 @@ public void initWithAccountStatus(
...
+        register(
+                "/v1/ais/" + userId,
+                200,
+                "application/json",
+                "{"
+                        + "  \"intervention\": {"
+                        + "    \"updatedAt\": 1696969322935,"
+                        + "    \"appliedAt\": 1696869005821,"
+                        + "    \"sentAt\": 1696869003456,"
+                        + "    \"description\": \"EXAMPLE_DESCRIPTION\""
```
[the remark this answers is not in the document]
Comment (isaac-GDS) [reply shared-test/src/main/java/uk/gov/di/authentication/sharedtest/extensions/AccountInterventionsStubExtension.java:109]: The answer above has resolved this. Happy for the acceptance test to capture the two states of response objects that can be returned

**Review by isaac-GDS — DISMISSED** [review]

**Review by isaac-GDS — APPROVED** [review]

**Pull request merged**